### PR TITLE
[enh] ensure that all apps in lists is ok

### PR DESCRIPTION
**Pull request by kay0u**
This script check the url, the branch an the revision for all apps in lists.

### CODE REVIEW

**Comment by maniackcrudelis** [comment]
@alexAubin @Psycojoker do you have any opinion on that ?

File: tests/testsRepo.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    jsonList = json.loads(fileList)
+
+    # No prompt for git
+    my_env = os.environ.copy()
+    my_env["GIT_TERMINAL_PROMPT"] = "0"
+
+    tempAppDir = "testedApp"
+
+    for app in jsonList:
+        url = jsonList[app]["url"]
```
[the remark this answers is not in the document]
Comment (kay0u) [reply tests/testsRepo.py]: Thank you! :+1:

**Comment by kay0u** [comment]
> Hmmm in fact I'm wondering how much of a good idea it is to run ~200 git clones in Travis ...
> 
> Github ain't gonna like that ... at least not without a token ... but even with a token it probably takes hell of a time :/
> 
> I dunno if there could be some trick in travis to only check apps affected by the commits ?

I can get modified lines in the PR (like a diff between the PR and master), retrieve the json part we want to check, and check it? It will be ugly but I can do it :-)

**Comment by alexAubin** [comment]
>I can get modified lines in the PR (like a diff between the PR and master), retrieve the json part we want to check, and check it? It will be ugly but I can do it :-)

Well yup, not sure if we can build a reliable way but anyway I doubt that we really want to trigger 250+ git clones each time we do a PR / commit here :P

**Comment by alexAubin** [comment]
What the hell, this is from more than 1 year ago !? I was convinced this was for less than 6 months ago x_x

Anyway, @kay0u : I know we love travis but what would you think about having our own cache of all the git clones (with --shallow-commit-whatever), and running the tests on our infra every day or X hours ? Not only is this less crazy in terms of having hundreds of git clone every commit (or weird engineering to test only the diff) but we could also : 
-  use it during the build process instead of having [forge-type dependant code logic to fetch info in the API](https://github.com/YunoHost/apps/blob/d235579e097215e6d389befd2042ca07c80e6dfb/list_builder.py#L162-L171)
- use it for analysis of helper usage or related stuff (had to do that kind of stuff a lot when working on helpers / package check) ...

**Comment by kay0u** [comment]
Meh 1 year already :/

I guess the list builder will fail if the sha1 is not valid for the repo? It might be enough.

> use it for analysis of helper usage or related stuff (had to do that kind of stuff a lot when working on helpers / package check) ...

The app CI is here for that right?

**Comment by alexAubin** [comment]
>The app CI is here for that right?

I was meaning thing like "finding out how many apps use helper X", or anything that involves to "grep -r" all apps repo ... or more recently running a new package_linter on all repo to find about all nginx path traversal issue, etc ...

Github sort of allow searches, but it's pretty limited

Anyway, I'm working on a rework of the list builder based on this idea right now ..

**Comment by jedie** [comment]
BTW. Travis is going to close support for opensouce projects, isn't it?!?

**Comment by ericgaspar** [comment]
We can close this PR I guess? Please reopen if I'm wrong

**Pull request closed without merging**